temporalcli/commands.taskqueue.go

```diff
@@ -11,31 +11,237 @@ import (
+	// Unversioned queue first
+	val, ok := taskQueueDescription.VersionsInfo[client.UnversionedBuildID]
+	if ok {
+		reachability, err := reachabilityToStr(val.TaskReachability)
+		if err != nil {
+			return nil, err
+		}
+		rRows = append(rRows, reachabilityRowType{
+			BuildID:      unversioned,
+			Reachability: reachability,
+		})
+	}
+	for k, val := range taskQueueDescription.VersionsInfo {
+		if k != client.UnversionedBuildID {
+			reachability, err := reachabilityToStr(val.TaskReachability)
+			if err != nil {
+				return nil, err
+			}
+			rRows = append(rRows, reachabilityRowType{
+				BuildID:      k,
+				Reachability: reachability,
+			})
+		}
+	}
+	return rRows, nil
+}
+
+func taskQueueTypeToStr(taskQueueType client.TaskQueueType) (string, error) {
+	switch taskQueueType {
+	case client.TaskQueueTypeUnspecified:
+		return "unspecified", nil
+	case client.TaskQueueTypeWorkflow:
+		return "workflow", nil
+	case client.TaskQueueTypeActivity:
+		return "activity", nil
+	case client.TaskQueueTypeNexus:
+		return "nexus", nil
+	default:
+		return "", fmt.Errorf("unrecognized task queue type: %d", taskQueueType)
+	}
+}
+
+func buildIDToPollerRows(pRows []pollerRowType, buildID string, typesInfo map[client.TaskQueueType]client.TaskQueueTypeInfo) ([]pollerRowType, error) {
+	for t, info := range typesInfo {
+		taskQueueType, err := taskQueueTypeToStr(t)
+		if err != nil {
+			return pRows, err
+		}
+		for _, p := range info.Pollers {
+			pRows = append(pRows, pollerRowType{
+				BuildID:        buildID,
+				TaskQueueType:  taskQueueType,
+				Identity:       p.Identity,
+				LastAccessTime: p.LastAccessTime,
+				RatePerSecond:  p.RatePerSecond,
+			})
+		}
+	}
+	return pRows, nil
+}
+
+func descriptionToPollerRows(taskQueueDescription client.TaskQueueDescription) ([]pollerRowType, error) {
+	var pRows []pollerRowType
+	var err error
+	// Unversioned queue first
+	val, ok := taskQueueDescription.VersionsInfo[client.UnversionedBuildID]
+	if ok {
+		pRows, err = buildIDToPollerRows(pRows, unversioned, val.TypesInfo)
+		if err != nil {
+			return nil, err
+		}
+	}
+	for k, val := range taskQueueDescription.VersionsInfo {
+		if k != client.UnversionedBuildID {
+			pRows, err = buildIDToPollerRows(pRows, k, val.TypesInfo)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return pRows, nil
+}
+
+func descriptionToRows(taskQueueDescription client.TaskQueueDescription) (taskQueueDescriptionType, error) {
+	rRows, err := descriptionToReachabilityRows(taskQueueDescription)
+	if err != nil {
+		return taskQueueDescriptionType{}, err
+	}
+	pRows, err := descriptionToPollerRows(taskQueueDescription)
+	if err != nil {
+		return taskQueueDescriptionType{}, err
+	}
+
+	return taskQueueDescriptionType{
+		Reachability: rRows,
+		Pollers:      pRows,
+	}, nil
+}
+
+func printTaskQueueDescription(cctx *CommandContext, taskQueueDescription client.TaskQueueDescription) error {
+	descRows, err := descriptionToRows(taskQueueDescription)
```

Comment:
Arguably limited value breaking these functions into other package-level functions that are only ever used once, but no biggie

Reply:
I tend to prefer smallish functions, but clearly they don't need to be package-level. I could add then as methods to the `TemporalTaskQueueDescribeCommand` object, but they are pure functions, ... What is the convention in Go for pure functions that are private to a file?

Reply:
There is no exact convention. Commonly they are inlined if they aren't very big and only called in one place.